web/src/hashchange.js

```diff
@@ -157,6 +159,22 @@ function do_hashchange_normal(from_reload) {
                 show_home_view();
                 return false;
             }
+
+            // Just select the message if the narrow is the same as the current narrow and
+            // we already have the new message we want to select rendered.
+            // TODO: Implement a filter comparison function so that we do it for any two narrows
+            // when calling narrow.activate.
```

Comment:
Yeah I agree with the thought that we should be doing this in `narrow.activate` if possible, so that if we add some helper click handler it still works as designed.